lib/client.js

```diff
@@ -1495,6 +1495,16 @@ function _resume (client, sync) {
   }
 }
 
+function shouldSendContentLength (contentLength, method) {
+  if (contentLength > 0) {
+    if (method === 'POST' || method === 'PUT' || method === 'PATCH') {
```

Comment:
Thinking twice is better if we exclude those who don't explicitly support the request's body `semantics`, so methods like `DELETE`, whose bodies are up to the server to decide what to do with them, can still have a `content-length` header.

Reply:
Made the changes, glad we're getting closer.